includes/base/controls-stack.php

```diff
@@ -2388,7 +2401,9 @@ protected function handle_control_position( array $args, $control_id, $overwrite
 				$args = array_replace_recursive( $target_tab, $args );
 			}
 		} elseif ( empty( $args['section'] ) && ( ! $overwrite || is_wp_error( Plugin::$instance->controls_manager->get_control_from_stack( $this->get_unique_name(), $control_id ) ) ) ) {
-			wp_die( sprintf( '%s::%s: Cannot add a control outside of a section (use `start_controls_section`).', get_called_class(), __FUNCTION__ ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			if ( ! $this->should_optimize_controls() ) {
```

Comment:
1. why do you need this condition? is there a case when you don't have a section that is valid in optimized environment?
2. what about cases where someone injects a control after a UI control?